docs/dp/basic.md

```diff
@@ -161,17 +161,45 @@ int dp() {
 
 当 $n$ 的范围扩大到 $n \leq 10^5$ 时，第一种做法就不够快了，下面给出了一个 $O(n \log n)$ 的做法。
 
-首先，定义 $a_1 \dots a_n$ 为原始序列，$d$ 为当前的不下降子序列，$len$ 为子序列的长度，那么 $d_{len}$ 就是长度为 $len$ 的不下降子序列末尾元素。
+首先，定义 $a_1 \dots a_n$ 为原始序列，$d_i$ 为所有的长度为 $i$ 的不下降子序列的末尾元素的最小值，$len$ 为子序列的长度。
 
 初始化：$d_1=a_1,len=1$。
 
-现在我们已知最长的不下降子序列长度为 1，那么我们让 $i$ 从 2 到 $n$ 循环，依次求出前 $i$ 个元素的最长不下降子序列的长度，循环的时候我们只需要维护好 $d$ 这个数组还有 $len$ 就可以了。**关键在于如何维护。**
+现在已知最长的不下降子序列长度为 1，那么我们让 $i$ 从 2 到 $n$ 循环，依次求出前 $i$ 个元素的最长不下降子序列的长度，循环的时候我们只需要维护好 $d$ 这个数组还有 $len$ 就可以了。**关键在于如何维护。**
 
 考虑进来一个元素 $a_i$：
 
 1.  元素大于等于 $d_{len}$，直接将该元素插入到 $d$ 序列的末尾。
 2.  元素小于 $d_{len}$，找到 **第一个** 大于它的元素，用 $a_i$ 替换它。
 
+为什么：
+
+首先，Case 1：
```

Comment:
```suggestion
首先，步骤 1：
```

https://oi-wiki.org/intro/format/#%E4%BE%8B-1

Reply:
fixed, thx.